### PR TITLE
Update to latest relay-api openapi spec

### PR DESCRIPTION
**Pull request by hunner**
Updates for at least the workflow folder handling.

The `WorkflowRevisions` endpoint contains a nested AllOf/OneOf that couldn't be worked around, so the build uses a 6.0.1-SNAPSHOT image of the recent master branch of openapi-generator-cli plus @impl's open PR #11837.

Placing `goimports` after the generation gives a 10x speedup to the `go generate` call.

### CODE REVIEW

File: .github/workflows/client-ci.yaml [hunk 1]
```diff
@@ -25,10 +25,14 @@ jobs:
         key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
         restore-keys: |
           ${{ runner.os }}-go-
+    - name: Install goimports
+      run: go get golang.org/x/tools/cmd/goimports
     - name: Check
       run: |
+        go mod tidy
```
Comment (hunner) [inline .github/workflows/client-ci.yaml:30]: Why is this needed? Can't figure it out.

File: client/hack/scripts/generate-jar [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+set -euo pipefail
+
+JAR_DIR="$(dirname "${BASH_SOURCE[0]}")"
+GENERATOR_VERSION="6.0.0"
```
Comment (hunner) [inline client/hack/scripts/generate-jar:5]: This file isn't needed any more, but was helpful when testing different local `.jar` builds without needing to make a new container.

Comment (rick-a-lane-ii) [reply client/hack/scripts/generate-jar:5]: I would prefer we avoid adding this file. A container-based and container-centric approach is preferable here.

File: client/pkg/client/openapi/api_workflow_runs.go [hunk 3]
```diff
@@ -12,74 +12,67 @@ package openapi
...
 // WorkflowRunsApiService WorkflowRunsApi service
 type WorkflowRunsApiService service
 
-type ApiGetWorkflowRunRequest struct {
-	ctx               _context.Context
+type WorkflowRunsApiGetWorkflowRunRequest struct {
+	ctx               context.Context
 	ApiService        *WorkflowRunsApiService
-	workflowName      string
+	workflowPath      GetWorkflowSubscriptionsWorkflowPathParameter
```
Comment (hunner) [inline client/pkg/client/openapi/api_workflow_runs.go:29]: When the `workflowPath.String` (which is a `*String`) is printed by `parameterToString` with `fmt.Sprintf("%v"...` it ends up returning the address rather than the value. I think this is another bug in openapi-generator. It seems to happen if a path parameter is a `OneOf`.

Comment (hunner) [reply client/pkg/client/openapi/api_workflow_runs.go]: I changed our openapi spec to work around this... just not using `OneOf` on path parameters for now.

**Review by kyleterry — APPROVED** [review]
LGTM

**Review by impl — APPROVED** [review]
LGTM
